x-pack/plugins/enterprise_search/server/routes/app_search/search.ts

```diff
@@ -36,4 +38,25 @@ export function registerSearchRoutes({
       path: '/api/as/v1/engines/:engineName/search.json',
     })
   );
+
+  // For the Search UI routes below, Search UI always uses the full API path, like:
+  // "/api/as/v1/engines/{engineName}/search.json". We only have control over the base path
+  // in Search UI, so we created a common basepath of "/api/app_search/search-ui" here that
+  // Search UI can use.
+  //
+  // Search UI *also* uses the click tracking and query suggestion endpoints, however, since the
+  // App Search plugin doesn't use that portion of Search UI, we only set up a proxy for the search endpoint below.
```

Comment:
From my own:

- It looks like Analytics search events are still getting logged
- However, Analytics click events are **not** getting logged, leading to inaccurate Analytics results when only using the dashboard
- It also it looks like API events are **no longer** getting logged, although this could be my local behaving strangely

@JasonStoltz I think the click event thing is potentially a blocker unfortunately, it prevents us from QAing Analytics accurately using the Documents view (important for things like E2E testing also). Is there any way we can update Search UI to either track clicks again, or (another option) to completely not affect Analytics whatsoever?

Reply:
Just want to add also: I personally don't think this fix needs to get in by 7.14FF (or it's also OK if we take a little longer and get it in past FF, since it's a bugfix). It's been broken on ent-search for several minors now, and this is shipping as a beta UI, so comes with even less support - so I'm OK taking a little longer to figure this out for the long term and to help solidify our E2E tests.

This may even be a decent time also to discuss/address the tech debt item we've discussed in the past about whether dashboard API calls should count towards Analytics/API events or not.

Reply:
> Sorry, I think my brain's a little fried this morning but just want to confirm I'm understanding this comment. By us not explicitly providing the click tracking and query suggestion routes in this PR, Search UI will no longer hit those endpoints. Is that correct?

Documents view is not configured to use click tracking or query suggestions. Since it never hits those endpoints, we do not need to create proxy endpoints here.

> It looks like Analytics search/query events are still getting logged

We should fix this, I don't think they should be logged

> However, Analytics click events are not getting logged, leading to inaccurate Analytics results when only using the dashboard

I think the endgame is to not log any of these queries or clicks.

> It also it looks like API events are no longer getting logged, although this could be my local behaving strangely

I don't know anything about that.

> I think the click event thing is potentially a blocker unfortunately, it prevents us from QAing Analytics accurately using the Documents view (important for things like E2E testing also).

That behavior isn't introduced in this PR, so I don't think this is a blocker for this PR. It is a separate issue entirely.

>  Is there any way we can update Search UI to either track clicks again, or (another option if that's easier) to completely not affect Analytics whatsoever?

I think the answer is the latter. Any views in the Dashboard that hit the search endpoint Query Tester, Curations, Documents View ... none of them should be logging Analytics events, they don't provide any value to the user. This is something we discussed as part of the Search UI view transition.

I'd propose a simple header that indicates "this request should not be tracked".

> Just want to add also: I personally don't think this fix needs to get in by 7.14FF (or it's also OK if we take a little longer and get it in past FF, since it's a bugfix). It's been broken on ent-search for several minors now, and this is shipping as a beta UI, so comes with even less support - so I'm OK taking a little longer to figure this out for the long term and to help solidify our E2E tests.

Just highlighting again that this PR is completely decoupled from that issue. If we want to enable click tracking we can, we just need to add a proxy endpoint for the click API route.

I think we should get this merged and fixed and handle analytics separately. I'm happy to create an issue for it in our backlog.

Reply:
> That behavior isn't introduced in this PR, so I don't think this is a blocker for this PR. It is a separate issue entirely.

To be clear, it is introduced by this PR (I'm not sure why though). It's not a separate issue unfortunately.

On Kibana master **and** on ent-search, you can do the following:

1. Go to the Documents view
2. Type in "park" (or any search that has results)
3. Click on any document
4. Go to Analytics
- [ ] Confirm that your "park" query shows up in Analytics in "Top queries" **and** in "Top queries with clicks" indicating that the click even triggered

On this branch, if you repeat the above 4 steps the "Top queries with clicks" panel does **not** update, indicating that click events stopped being recorded.

Unfortunately, even if this is a side effect of this PR, it's not one I feel 100% comfortable creating so close to FF since it's a distinct deviation from how ent-search currently behaves. FWIW, I do think the best long term solution is to not have any dashboard actions show up in either Analytics or API logs - I'd just like to take the extra effort to get that in if possible 🤔

Reply:
> I think we should get this merged and fixed and handle analytics separately. I'm happy to create an issue for it in our backlog.

Sorry, should have addressed this too. I'm OK with merging this in now only if we commit to fully fixing analytics **before** 7.14 (so I guess after FF but before release). Otherwise, I think we should wait for 7.15. I really really worry about releasing this to 7.14 with an entire Analytics category (queries with clicks) incorrectly affected by our own dashboard in a way that it has not done so previously.

Would definitely appreciate it if you could try to repro my above steps and confirm behavior in master vs this branch!

Reply:
I think you're completely misunderstanding me.

We do not, and never have tracked click analytics in the Documents view within Kibana. This PR does not change that. It doesn't prevent us from doing it in the future either. This PR is completely decoupled from your Analytics concern.

Whether we merge this PR or don't, it has no impact on Analytics.

Reply:
I cannot reproduce.

Both on master kibana and this branch, clicking on a link will not send click data.

Inspect the network tab and you'll see for yourself, no API call is being made.

We *did* track these clicks in ent-search.

We *do not*, and never have tracked click analytics in the Documents view within Kibana. This PR does not change that. It doesn't prevent us from doing it in the future either. This PR is completely decoupled from your Analytics concern.

Whether we merge this PR or don't, it has no impact on Analytics.

Reply:
Gah, yup, you're totally right. I just tried again on Kibana master with a fresh instance and clicks aren't showing up. Not sure if I was in a horrible fever dream, if my local elasticsearch instance got messed up, or if I got confused with `ent-search` 😰 (or all of the above?)

Thanks so much for taking the extra time to repro and confirm for me, I really appreciate that. Going to ahead and approve, and bring up filtering out dashboard analytics at next week's sync

Reply:
After some thought I feel like the above comment does not convey my feeling of "Sorry for being totally wrong" enough so I'll fall back to my normal communication crutch of funny gifs

<img src="https://media.giphy.com/media/1Bh2zyW2yuBEalV0rO/giphy.gif" alt="" width="300">

Reply:
Haha, thanks for calling it out. Better to call it out and be wrong than say nothing.

We can definitely solve this Analytics thing. Maybe just enable click analytics for 7.14 and remove all analytics for 7.15.